grpc-gcp/src/gcp_channel_factory.ts

```diff
@@ -263,7 +264,19 @@ export class GcpChannelFactory {
     deadline: grpc.Deadline,
     callback: Function
   ): void {
-    throw new Error('Function watchConnectivityState not implemented!');
+    if (!this.channelRefs.length) {
```

Comment:
I actually don't know if we can get into this state? This might be unnecessary.

Reply:
Is there any way you can fake the behavior in test, at least so we confirm the error _would_ be handled as expected?

Reply:
I'm still getting into the tests, but I'm going to say yes, absolutely!

Reply:
By design the channel factory constructor will create an underlying channel to avoid empty channel pool. But it would be good to have this checker. Also thanks for adding the tests.